File: scripts/test-upload.js

const fs = require('fs');
const path = require('path');
require('dotenv').config({ path: '.env.local' });

const apiKey = process.env.MAGIC_HOUR_API_KEY;
const baseUrl = 'https://api.magichour.ai/v1';
const logFile = 'probe_results.txt';

function log(msg) {
    console.log(msg);
    fs.appendFileSync(logFile, msg + '\n');
}

async function testUploadAndGen() {
    fs.writeFileSync(logFile, '--- Testing Magic Hour Lip Sync (Final Proof) ---\n');
    log('Starting...');

    // 1. Get arguments
    const filePath = path.join(process.cwd(), 'public', 'ai_avatar', 'sarah.png');
    if (!fs.existsSync(filePath)) return;

    // 2. Upload
    let fileId;
    try {
        const payload = { items: [{ extension: 'png', type: 'image' }] };
        const res = await fetch(`${baseUrl}/files/upload-urls`, {
            method: 'POST',
            headers: { 'Authorization': `Bearer ${apiKey}`, 'Content-Type': 'application/json' },
            body: JSON.stringify(payload)
        });
        const data = await res.json();
        const uploadInfo = data.items[0];
        fileId = uploadInfo.file_path || uploadInfo.key || uploadInfo.id;

        const upUrl = uploadInfo.url || uploadInfo.upload_url || uploadInfo.signed_url;
        const fileBuffer = fs.readFileSync(filePath);
        await fetch(upUrl, {
            method: 'PUT',
            headers: { 'Content-Type': 'image/png' },
            body: fileBuffer
        });
        log(`✅ Uploaded input image. File Path: ${fileId}`);
    } catch (e) {
        log('Upload Error: ' + e.message);
        return;
    }

    // 3. Final Probe
    // We use a known public audio URL
    const audioUrl = "https://actions.google.com/sounds/v1/alarms/beep_short.ogg";

    log(`\nTesting /lip-sync with Image + URL Audio...`);
    try {
        const body = {
            start_seconds: 0,
            end_seconds: 5,
            assets: {
                video_source: "file",
                video_file_path: fileId,
                audio_file_path: audioUrl
            }
        };

        const gRes = await fetch(`${baseUrl}/lip-sync`, {
            method: 'POST',
            headers: {
                'Authorization': `Bearer ${apiKey}`,
                'Content-Type': 'application/json'
            },
            body: JSON.stringify(body)
        });

        log(`Status: ${gRes.status}`);
        const text = await gRes.text();
        log(`Response: ${text.substring(0, 500)}`);

    } catch (e) {
        log('Probe Error: ' + e.message);
    }
}

testUploadAndGen();

File: scripts/test_upload.py
import requests
import os

# Custom .env loader
def load_env():
    try:
        with open('.env.local', 'r') as f:
            for line in f:
                line = line.strip()
                if '=' in line and not line.startswith('#'):
                    # Split only on first =
                    parts = line.split('=', 1)
                    key = parts[0].strip()
                    value = parts[1].strip()
                    # Handle quotes
                    if value.startswith('"') and value.endswith('"'):
                        value = value[1:-1]
                    elif value.startswith("'") and value.endswith("'"):
                        value = value[1:-1]
                    os.environ[key] = value
    except Exception as e:
        print(f"Error loading .env: {e}")

load_env()
api_key = os.environ.get('HEYGEN_API_KEY')
print(f"API Key loaded: {'Yes' if api_key else 'No'}")

# Test variants
file_path = "public/ai_avatar/sarah.png"
# Final test: Auto-detect mime type
print(f"\n--- Testing Auto-Detect MIME ---")
try:
    with open(file_path, 'rb') as f:
        # Let requests guess the MIME type
        files = {'file': f}
        headers = {'X-Api-Key': api_key}
        
        response = requests.post("https://upload.heygen.com/v1/asset", headers=headers, files=files)
        
        print(f"Status: {response.status_code}")
        print(f"Body: {response.text}")

except Exception as e:
    print(f"Error: {e}")
